### ISSUE TITLE
Logfile being spammed with "NoSuchFileException"

### ISSUE THREAD

**Issue by Cyberuben** [elastic/elasticsearch#15734]
As I was just trying to access my ElasticSearch instance, I got prompted with several 503 errors. After accessing the server and trying to check the logs, I saw that my logfile was 1.8 GB big and consisted mainly of errors such as the following:

```
[2016-01-02 13:56:18,601][WARN ][cluster.action.shard     ] [Brynocki] [admin_dashboard][0] received shard failed for [admin_dashboard][0], node[hIMImTGASSmIHX_bUAVa8w], [P], v[39], s[INITIALIZING], a[id=c51WUYPuSIOiSe2Pox-SHA], unassigned_info[[reason=ALLOCATION_FAILED], at[2016-01-02T12:56:15.006Z], details[failed recovery, failure IndexShardRecoveryException[failed to recovery from gateway]; nested: EngineCreationFailureException[failed to recover from translog]; nested: EngineException[failed to recover from translog]; nested: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833]; ]], indexUUID [XE89emCJQeOLqRItz7LAgw], message [failed recovery], failure [IndexShardRecoveryException[failed to recovery from gateway]; nested: EngineCreationFailureException[failed to recover from translog]; nested: EngineException[failed to recover from translog]; nested: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833]; ]
[admin_dashboard][[admin_dashboard][0]] IndexShardRecoveryException[failed to recovery from gateway]; nested: EngineCreationFailureException[failed to recover from translog]; nested: EngineException[failed to recover from translog]; nested: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833];
        at org.elasticsearch.index.shard.StoreRecoveryService.recoverFromStore(StoreRecoveryService.java:254)
        at org.elasticsearch.index.shard.StoreRecoveryService.access$100(StoreRecoveryService.java:56)
        at org.elasticsearch.index.shard.StoreRecoveryService$1.run(StoreRecoveryService.java:129)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: [admin_dashboard][[admin_dashboard][0]] EngineCreationFailureException[failed to recover from translog]; nested: EngineException[failed to recover from translog]; nested: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833];
        at org.elasticsearch.index.engine.InternalEngine.<init>(InternalEngine.java:178)
        at org.elasticsearch.index.engine.InternalEngineFactory.newReadWriteEngine(InternalEngineFactory.java:25)
        at org.elasticsearch.index.shard.IndexShard.newEngine(IndexShard.java:1408)
        at org.elasticsearch.index.shard.IndexShard.createNewEngine(IndexShard.java:1403)
        at org.elasticsearch.index.shard.IndexShard.internalPerformTranslogRecovery(IndexShard.java:906)
        at org.elasticsearch.index.shard.IndexShard.performTranslogRecovery(IndexShard.java:883)
        at org.elasticsearch.index.shard.StoreRecoveryService.recoverFromStore(StoreRecoveryService.java:245)
        ... 5 more
Caused by: [admin_dashboard][[admin_dashboard][0]] EngineException[failed to recover from translog]; nested: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833];
        at org.elasticsearch.index.engine.InternalEngine.recoverFromTranslog(InternalEngine.java:254)
        at org.elasticsearch.index.engine.InternalEngine.<init>(InternalEngine.java:175)
        ... 11 more
Caused by: TranslogCorruptedException[translog corruption while reading from stream]; nested: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833];
        at org.elasticsearch.index.translog.Translog.readOperation(Translog.java:1636)
        at org.elasticsearch.index.translog.TranslogReader.read(TranslogReader.java:132)
        at org.elasticsearch.index.translog.TranslogReader$ReaderSnapshot.readOperation(TranslogReader.java:299)
        at org.elasticsearch.index.translog.TranslogReader$ReaderSnapshot.next(TranslogReader.java:290)
        at org.elasticsearch.index.translog.MultiSnapshot.next(MultiSnapshot.java:70)
        at org.elasticsearch.index.engine.InternalEngine.recoverFromTranslog(InternalEngine.java:240)
        ... 12 more
Caused by: TranslogCorruptedException[translog stream is corrupted, expected: 0xc1ef600a, got: 0x30373833]
        at org.elasticsearch.index.translog.Translog.verifyChecksum(Translog.java:1593)
        at org.elasticsearch.index.translog.Translog.readOperation(Translog.java:1626)
        ... 17 more
[2016-01-02 13:56:18,813][WARN ][index.translog           ] [Brynocki] [admin_dashboard][0] failed to delete temp file /var/lib/elasticsearch/onlyonce_statistics/nodes/0/indices/admin_dashboard/0/translog/translog-3301628599119632219.tlog
java.nio.file.NoSuchFileException: /var/lib/elasticsearch/onlyonce_statistics/nodes/0/indices/admin_dashboard/0/translog/translog-3301628599119632219.tlog
        at sun.nio.fs.UnixException.translateToIOException(Unknown Source)
        at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
        at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
        at sun.nio.fs.UnixFileSystemProvider.implDelete(Unknown Source)
        at sun.nio.fs.AbstractFileSystemProvider.delete(Unknown Source)
        at java.nio.file.Files.delete(Unknown Source)
        at org.elasticsearch.index.translog.Translog.recoverFromFiles(Translog.java:324)
        at org.elasticsearch.index.translog.Translog.<init>(Translog.java:166)
        at org.elasticsearch.index.engine.InternalEngine.openTranslog(InternalEngine.java:209)
        at org.elasticsearch.index.engine.InternalEngine.<init>(InternalEngine.java:152)
        at org.elasticsearch.index.engine.InternalEngineFactory.newReadWriteEngine(InternalEngineFactory.java:25)
        at org.elasticsearch.index.shard.IndexShard.newEngine(IndexShard.java:1408)
        at org.elasticsearch.index.shard.IndexShard.createNewEngine(IndexShard.java:1403)
        at org.elasticsearch.index.shard.IndexShard.internalPerformTranslogRecovery(IndexShard.java:906)
        at org.elasticsearch.index.shard.IndexShard.performTranslogRecovery(IndexShard.java:883)
        at org.elasticsearch.index.shard.StoreRecoveryService.recoverFromStore(StoreRecoveryService.java:245)
        at org.elasticsearch.index.shard.StoreRecoveryService.access$100(StoreRecoveryService.java:56)
        at org.elasticsearch.index.shard.StoreRecoveryService$1.run(StoreRecoveryService.java:129)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```

I can't seem to find out how to recover from this problem

**Comment by jpountz** [comment]
What version are you running? I'm thinking it could be #14695

**Comment by Cyberuben** [comment]
That'd be the following:

```
"version": {
    "number": "2.1.0",
    "build_hash": "72cd1f1a3eee09505e036106146dc1949dc5dc87",
    "build_timestamp": "2015-11-18T22:40:03Z",
    "build_snapshot": false,
    "lucene_version": "5.3.1"
  }
```

**Comment by jpountz** [comment]
The "failed to delete temp file" part would be addressed by #14872 I think. However the other messages look to me like there is a genuine corruption of the translog file. @s1monw does it ring a bell to you?

**Comment by clintongormley** [comment]
Did you have a disk full event? I think it is related to https://github.com/elastic/elasticsearch/issues/15333

**Comment by Cyberuben** [comment]
The disk was full after this spammed my log file, yes, but I don't think a full disk was the cause. I had roughly 16GB of log files, and I only had 20GB diskspace.

**Comment by s1monw** [comment]
This was fixed in 2.1.1 as clint said https://github.com/elastic/elasticsearch/issues/15333 is the relevant issue. The log spamming is also fixed  #14872 - I will close it now since all issues have been resolved in 2.1.1 and 2.0.2 @Cyberuben thanks for reporting this it's super important to get as much info as possible if stuff like this happens.
